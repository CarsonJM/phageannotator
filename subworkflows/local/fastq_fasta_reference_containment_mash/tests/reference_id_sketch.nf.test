nextflow_workflow {

    name "Test Workflow main.nf"
    script "subworkflows/local/reference_virus_identification/main.nf"
    workflow "REFERENCE_VIRUS_IDENTIFICATION"
    tag "subworkflows_local"
    tag "reference_virus_identification"
    tag "reference_id_sketch_provided"


    test("params.reference_id_sketch provided") {

        when {
            params {
                outdir              = "$outputDir"
                reference_id_fasta  = 'https://github.com/nf-core/test-datasets/raw/modules/data/genomics/prokaryotes/bacteroides_fragilis/genome/genome.fna.gz'
                reference_id_sketch = 'https://github.com/nf-core/test-datasets/raw/phageannotator/subworkflows/local/reference_virus_identification/bacteroides_fragilis.msh'
            }
            workflow {
                """
                input[0] = [ [ id:'test1' ], [file('https://github.com/nf-core/test-datasets/raw/modules/data/genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_1.fastq.gz', checkIfExists: true), file('https://github.com/nf-core/test-datasets/raw/modules/data/genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_2.fastq.gz', checkIfExists: true)]]
                input[1] = [ [ id:'test1' ], file('https://github.com/nf-core/test-datasets/raw/modules/data/genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz', checkIfExists: true)]
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert workflow.trace.succeeded().size() == 2 },
                { assert snapshot(path("$outputDir/ReferenceIdentification/mash/screen/reference.screen")).match("mash_screen") },
                { assert snapshot(path("$outputDir/ReferenceIdentification/append_screen_hits/test1.fasta_w_screen_hits.fna.gz")).match("append_screen_hits") }
            )
        }
    }
}
