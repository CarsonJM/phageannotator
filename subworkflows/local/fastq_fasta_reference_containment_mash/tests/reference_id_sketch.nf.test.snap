{
    "mash_screen": {
        "content": [
            "reference.screen:md5,958af1f9eb156ebd2978e0596ae4eda0"
        ],
        "timestamp": "2023-08-09T20:59:28+0000"
    },
    "append_screen_hits": {
        "content": [
            "test1.fasta_w_screen_hits.fna.gz:md5,c249a103f8836ec32cb837ca167db162"
        ],
        "timestamp": "2023-08-09T20:59:28+0000"
    }
}